internal/http/download.go

```diff
@@ -36,23 +37,35 @@ type DownloadOption func(*downloadOptions)
 func Download(ctx context.Context, url string, target io.Writer, options ...DownloadOption) (err error) {
 	opts := downloadOptions{
 		stalenessTimeout: time.Minute,
+		header:           http.Header{},
 	}
 	for _, opt := range options {
 		opt(&opts)
 	}
 
-	// Prepare the client and the request.
-	client := http.Client{
-		Transport: &http.Transport{
-			// This is a one-shot HTTP client which should release resources immediately.
-			DisableKeepAlives: true,
-		},
+	transport := http.DefaultTransport.(*http.Transport).Clone()
```

Comment:
It was a deliberate decision not to use the default transport here, as it defines a bunch of timeouts that make configuration difficult. I think for this simple use case the staleness timeout is all that's needed.

The only thing that's probably good to set here is `ProxyFromEnvironment`. I'd even say setting this here should actually be backported.

Reply:
https://github.com/k0sproject/k0s/pull/5064